lighthouse-core/gather/gather-runner.js

```diff
@@ -187,17 +206,56 @@ class GatherRunner {
       const status = `Retrieving: ${gatherer.name}`;
       return chain.then(_ => {
         log.log('status', status);
-        return gatherer.afterPass(options, passData);
-      }).then(ret => {
+        const artifactPromise = Promise.resolve().then(_ => gatherer.afterPass(options, passData));
+        gathererResults[gatherer.name].push(artifactPromise);
+        return artifactPromise.catch(err => {
+          if (!err.recoverable) {
```

Comment:
(this is why I still lean toward a consistent `RecoverableError` constructor we can use. totally don't care if it's an actual subclass or just hacks `.name` and adds a `.recoverable = true` :) )